src/backend/distributed/utils/metadata_cache.c

```diff
@@ -155,6 +156,7 @@ static HTAB *DistTableCacheHash = NULL;
 
 /* Hash table for informations about each shard */
 static HTAB *DistShardCacheHash = NULL;
+static MemoryContext ShardCacheMemoryContext = NULL;
```

Comment:
Maybe `MetadataCacheMemoryContext`? the hash table used to cache distributed table information is also allocated in this memory context.

Optional, but it would be good to use this in place of CacheMemoryContext whenever we allocate something into the cache.

Reply:
done